### PR TITLE
txnengine, txnstorage: log tail

### PR DESCRIPTION
**Pull request by reusee**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:
txnengine: add GetLogTail
txnstorage: add GetLogTail

### CODE REVIEW

File: pkg/txn/storage/txn/log_tail.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+			if value.LockTx != nil &&
+				value.LockTx.State.Load() == Committed &&
+				value.LockTime.After(fromTime) &&
+				value.LockTime.Before(toTime) {
+				// committed delete
+				namedRow := &NamedAnyRow{
+					Row:      value.Value,
+					AttrsMap: attrsMap,
+				}
+				if err := appendNamedRow(tx, m.mheap, deleteBatch, namedRow); err != nil {
```
Comment (nnsgmsone) [inline pkg/txn/storage/txn/log_tail.go:114]: When is the memory released? I don't see the code to release it

Comment (reusee) [reply pkg/txn/storage/txn/log_tail.go:131]: fixed

**Review by nnsgmsone — APPROVED** [review]

File: pkg/vm/engine/txn/log_tail.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+		func() ([]Shard, error) {
+			return t.engine.shardPolicy.Stores([]logservicepb.DNStore{targetStore})
+		},
+		OpGetLogTail,
+		GetLogTailReq{
+			TableID: t.id,
+			Request: apipb.SyncLogTailReq{
+				CnHave: from,
+				CnWant: to,
+				Table:  &apipb.TableID{}, // we dont use this
```
Comment (nnsgmsone) [inline pkg/vm/engine/txn/log_tail.go:48]: This is necessary

Comment (reusee) [reply pkg/vm/engine/txn/log_tail.go:48]: Table id is passed by TableID of GetLogTailReq

**Review by fengttt — APPROVED** [review]

**Pull request merged**